layer/trace_layer.cpp

```diff
@@ -337,9 +389,40 @@ VKAPI_ATTR VkResult VKAPI_CALL EnumerateDeviceExtensionProperties(VkPhysicalDevi
     {
         // If this function was not called with the layer's name, we expect to dispatch down the chain to obtain the ICD
         // provided extensions.
-        result = encode::GetInstanceTable(physicalDevice)
-                     ->EnumerateDeviceExtensionProperties(
-                         encode::GetWrappedHandle(physicalDevice), pLayerName, pPropertyCount, pProperties);
+        // In order to screen out blacklisted unsupported extensions, we always query
+        // the chain twice, and remove those that are present from the count.
+        auto     instance_table            = encode::GetInstanceTable(physicalDevice);
+        auto     wrapped_device            = encode::GetWrappedHandle(physicalDevice);
+        uint32_t downstream_property_count = 0;
+
+        result = instance_table->EnumerateDeviceExtensionProperties(
+            wrapped_device, pLayerName, &downstream_property_count, nullptr);
+        if (result != VK_SUCCESS)
+            return result;
+
+        std::vector<VkExtensionProperties> downstream_properties;
```

Comment:
Can this and the next line be combined into one of the constructor forms that takes the size?

Reply:
Done. Bad habit.